subsys/wifi/shell/shell.c

```diff
@@ -0,0 +1,626 @@
+				    status->u.sta.host_rssi);
+		}
+	} else if (!strcmp(iface_name, WIFIMGR_IFACE_NAME_AP)) {
+		shell_print(shell, "AP Status:\t%s", ap_sts2str(status->state));
+		if (status->own_mac && !is_zero_ether_addr(status->own_mac))
+			shell_print(shell, "BSSID:\t\t" MACSTR "",
+				    MAC2STR(status->own_mac));
+
+		if (status->state == WIFI_STATE_AP_STARTED) {
+			int i;
+			char (*mac_addrs)[WIFI_MAC_ADDR_LEN];
+
+			shell_print(shell, "----------------");
+			shell_print(shell, "STA NR:\t%u", status->u.ap.nr_sta);
+			if (status->u.ap.nr_sta) {
+				shell_print(shell, "STA:");
+				mac_addrs = status->u.ap.sta_mac_addrs;
+				for (i = 0; i < status->u.ap.nr_sta; i++)
+					shell_print(shell, "\t\t" MACSTR "",
+						    MAC2STR(mac_addrs[i]));
+			}
+
+			shell_print(shell, "----------------");
+			shell_print(shell, "ACL NR:\t%u", status->u.ap.nr_acl);
+			if (status->u.ap.nr_acl) {
+				shell_print(shell, "ACL:");
+				mac_addrs = status->u.ap.acl_mac_addrs;
+				for (i = 0; i < status->u.ap.nr_acl; i++)
+					shell_print(shell, "\t\t" MACSTR "",
+						    MAC2STR(mac_addrs[i]));
+			}
+		}
+	}
+}
+
+static void wifimgr_cli_show_scan_res(struct wifi_scan_result *scan_res)
+{
+	if (scan_res->ssid && strlen(scan_res->ssid))
+		printf("\t%-32s", scan_res->ssid);
+	else
+		printf("\t\t\t\t\t");
+
+	if (scan_res->bssid && !is_zero_ether_addr(scan_res->bssid))
+		printf("\t" MACSTR, MAC2STR(scan_res->bssid));
+	else
+		printf("\t\t\t");
+
+	printf("\t%s", security2str(scan_res->security));
+	printf("\t%u\t%d\n", scan_res->channel, scan_res->rssi);
+}
+
+#ifdef CONFIG_WIFIMGR_STA
+static void wifimgr_cli_show_rtt_resp(struct wifi_rtt_response *rtt_resp)
+{
+	if (rtt_resp->bssid && !is_zero_ether_addr(rtt_resp->bssid))
+		printf("\t" MACSTR, MAC2STR(rtt_resp->bssid));
+	else
+		printf("\t\t\t");
+
+	printf("\t%d\n", rtt_resp->range);
+}
+#endif
+
+static int strtomac(char *mac_str, char *mac_addr)
+{
+	char *mac;
+	int i;
+
+	mac = strtok(mac_str, ":");
+
+	for (i = 0; i < WIFI_MAC_ADDR_LEN; i++) {
+		char *tail;
+
+		mac_addr[i] = strtol(mac, &tail, 16);
+		mac = strtok(NULL, ":");
+
+		if (!mac)
+			break;
+	}
+
+	if (i != (WIFI_MAC_ADDR_LEN - 1))
+		return -EINVAL;
+
+	return 0;
+}
+
+/* WiFi Manager CLI client commands */
+static int wifimgr_cli_cmd_set_config(const struct shell *shell, size_t argc,
+				      char *argv[])
+{
+	struct wifi_config conf;
+	char *iface_name;
+	int choice;
+
+	if (!argv[1])
+		return -EINVAL;
+	iface_name = argv[1];
+
+	if (!strcmp(iface_name, WIFIMGR_IFACE_NAME_STA))
+		shell_print(shell, "Setting STA Config ...");
+	else if (!strcmp(iface_name, WIFIMGR_IFACE_NAME_AP))
+		shell_print(shell, "Setting AP Config ...");
+	else
+		return -EINVAL;
+
+	memset(&conf, 0, sizeof(conf));
+
+	optind = 0;
+	while ((choice = getopt(argc, argv, "a:b:c:m:n:p:w:")) != -1) {
+		switch (choice) {
+		case 'a':
+			conf.autorun = atoi(optarg);
+			if (!conf.autorun) {
+				shell_print(shell, "invalid autorun!");
```

Comment:
Just a proposal. It would be more neat to use `shell_error` in error messages.

Reply:
@jarz-nordic
Done, thanks!